crates/blockifier/src/execution/contract_class.rs

```diff
@@ -542,69 +545,184 @@ impl Deref for NativeContractClassV1 {
     }
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum NativeEntryPointError {
+    #[error("FunctionId {0} not found")]
+    FunctionIdNotFound(usize),
+}
+
 impl NativeContractClassV1 {
     fn constructor_selector(&self) -> Option<EntryPointSelector> {
-        let entrypoint = self.deref().entry_points_by_type.constructor.first()?;
-        Some(contract_entrypoint_to_entrypoint_selector(entrypoint))
+        self.entry_points_by_type.constructor.first().map(|ep| ep.selector)
     }
 
-    pub fn try_from_json_string(raw_contract_class: &str) -> Result<Self, ProgramError> {
-        fn compile(sierra_program: &SierraProgram) -> AotNativeExecutor {
-            let native_context = cairo_native::context::NativeContext::new();
-            let native_program = native_context.compile(sierra_program, None).unwrap();
-            AotNativeExecutor::from_native_module(native_program, OptLevel::Default)
-        }
-
-        let sierra_contract_class: SierraContractClass = serde_json::from_str(raw_contract_class)?;
-        // todo(rodro): we are having two instances of a sierra program, one it's object form
-        // and another in its felt encoded form. This can be avoided by either:
-        //   1. Having access to the encoding/decoding functions
-        //   2. Refactoring the code on the Cairo mono-repo
-
-        let sierra_program = sierra_contract_class.extract_sierra_program().unwrap();
-        let executor = compile(&sierra_program);
+    /// Initialize a compiled contract class for native.
+    ///
+    /// executor must be derived from sierra_program which in turn must be derived from
+    /// sierra_contract_class.
+    pub fn new(
+        sierra_program: &SierraProgram,
+        executor: AotNativeExecutor,
+        sierra_contract_class: SierraContractClass,
```

Comment:
We can get the Sierra program for the Sierra Contract Class (with `extract_sierra_program`) so I think the first argument might be redundant